aws/aws-parallelcluster
# [3.x] Add a Resource-Prefix Config Option to CLI

### Description of changes

* Providing Resource Prefix config option so that user is able to customize the IAM resource's path prefix and role based prefix.
* Changing schema `ClusterIamSchema` and resource `ClusterIam` to enable user to give ResourcePrefix config option.
* Using IAM naming conventions to create a Regex which will validate the input.

### Tests
* Manual Tests

      1. Created a cluster with wrong Resource Prefix format to see the validation ERROR
      2. Created a cluster with wrong Resource Prefix format and suppressed validators to see the no validation ERROR
      3. Created a Cluster with correct Resource Prefix format (this wont be used for resource creation yet)

* Automated Tests

      1. Unit test cases to validate the Regex handles both positive and negative test scenarios.

### References
* [IAM Naming Conventions](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html)

### Checklist

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
